Rename MemSQL docs to SingleStore

colebow opened the pull request:
<!-- Thank you for submitting a pull request! Find more information in our development guide at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md and contact us on #dev in Slack. -->
<!-- Provide an overview of the PR for maintainers and reviewers. -->
## Description

We've fully deprecated the memsql name, so we should change the name of the .rst file so it appears as "singlestore.html" in the browser URL. Tested & built locally, doesn't seem as if anything else links to this page.

<!-- Provide details that would help an engineer who is unfamiliar with this part of the code. -->
## Additional context and related issues

Follow-up to #16180 

<!-- Mark the appropriate option with an (x). Propose a release note if you can. -->
## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

ebyhr reviewed (commented):
> doesn't seem as if anything else links to this page.

What about external usages (not managed in this repository)? We don't care about it? Relates to #15467

mosabua commented:
Yeah .. we want to look at the redirect stuff in parallel

colebow commented:
> > doesn't seem as if anything else links to this page.
> 
> I didn't rename this page intentionally because I have concerns about external usages (=not managed in this repository). We don't care about it? Relates to #15467

It's possible some pages aren't indexed by google, but I can't find an external link to our memsql/singlestore docs at all.

colebow commented:
@electrum @jhlodin borrowing from #15467, I've added a redirect plugin to this PR. To allow the build to function for now, I've removed the "-W" flag, which no longer treats build warnings as errors. [I've opened up a pull request for the plugin](https://github.com/sphinx-contrib/redirects/pull/6) to fix the warning, and once that's included, we can re-add the -W flag.

jhlodin commented:
Note that this would require a new Sphinx docker image (with the new redirects plugin) to be pushed to `ghcr.io`, which @electrum handled when we upgraded Sphinx to 5.0.2 last year.

electrum commented on docs/requirements.txt:
```diff
@@ -1,87 +1,81 @@
 #
```
Is the commit description wrong? This doesn't seem like a revert

mosabua commented on docs/src/main/sphinx/ext/redirects.py:
```diff
@@ -0,0 +1,86 @@
+#   :license: BSD, see LICENSE for details.
+#   https://github.com/sphinx-contrib/redirects/blob/master/LICENSE
+#
+#   Modified April 10, 2023 for Trino use
```
How? Maybe link to issue and also detail what version was modified .. so we can figure it out a bit easier in the future

ebyhr commented on docs/src/main/sphinx/ext/redirects.py:
```diff
@@ -0,0 +1,87 @@
+#   https://github.com/sphinx-contrib/redirects/blob/master/LICENSE
+#
+#   Version 0.1.0 modified April 10, 2023 for Trino use
+#   See https://github.com/sphinx-contrib/redirects/pull/6
```
This PR is already merged. Why not wait for the next release?

mosabua replied:
We asked for a release .. seems like they team is not responding at this stage.. hence we pulled the extension in for now.. once a release becomes available we will switch bak to the official extension

ebyhr reviewed (commented):
The commit order is wrong. "Add redirects extension ..." should be the 1st.

mosabua reviewed (approved):
Looks good now. Lets make sure we check upstream now and then and see if they cut a release .. then we can get rid of the inline redirect and build a new container image and use that.
